mos-platform/common/lib/new.cc

```diff
@@ -311,6 +367,35 @@ __attribute__((weak)) void *malloc(size_t count) {
   return free_list.split_block(free_list.find_first_fit(count), count);
 }
 
+__attribute__((weak)) void *realloc(void *orig, size_t count) {
+  if (!orig) {
+    return malloc(count);
+  }
+
+  const auto orig_block_ptr = block::get_block(static_cast<std::byte *>(orig));
+  if (count > orig_block_ptr->m_size) {
+    auto &free_list = get_free_list();
+    const auto grow_sz = count - orig_block_ptr->m_size;
+
+    // Find the adjacent block, and if found split it.
+    const auto adjacent_block =
+        free_list.find_adjacent_fit(orig_block_ptr, grow_sz);
+    if (adjacent_block) {
```

Comment:
Similarly, `if (!adjacent_block) ... return`, then the remainder can be dedented.